src/controller/CHIPDeviceController.cpp

```diff
@@ -1523,16 +1544,21 @@ void DeviceCommissioner::OnDeviceConnectedFn(void * context, OperationalDevicePr
             commissioner->CommissioningStageComplete(CHIP_NO_ERROR, report);
         }
     }
-    else if (commissioner->mPairingDelegate != nullptr)
-    {
-        commissioner->mPairingDelegate->OnPairingComplete(CHIP_NO_ERROR);
-    }
-
-    // Release any CommissioneeDeviceProxies we have here as we now have an OperationalDeviceProxy.
-    CommissioneeDeviceProxy * commissionee = commissioner->FindCommissioneeDevice(device->GetDeviceId());
-    if (commissionee != nullptr)
+    else
     {
-        commissioner->ReleaseCommissioneeDevice(commissionee);
+        if (commissioner->mPairingDelegate != nullptr)
+        {
+            commissioner->mPairingDelegate->OnPairingComplete(CHIP_NO_ERROR);
+        }
+        // Only release the pase session if we're not commissioning. If we're commissioning, we're going to hold onto that pase
+        // session until we send the commissioning complete command just in case it fails and we need to go back to the pase
+        // connection to re-setup the network. This is unlikely, given that we just connected over the operational network, but is
```

Comment:
```suggestion
        // Only release the PASE session if we're not commissioning. If we're commissioning, we're going to hold onto that PASE
        // session until we send the commissioning complete command just in case it fails and we need to go back to the PASE
        // connection to re-setup the network. This is unlikely, given that we just connected over the operational network, but is
```